examples/Pages.elm

```diff
@@ -0,0 +1,212 @@
+{- This example app shows how to use navigation functionality for showing
+   different "pages" (views) based on url. Updates to url render different page views.
+-}
+
+
+module Main exposing (..)
+
+import Html exposing (..)
+import Html.Attributes exposing (style, href, src)
+import Html.Events exposing (..)
+import Navigation
+import String
+
+
+main : Program Never
+main =
+  Navigation.program urlParser
+    { init = init
+    , view = view
+    , update = update
+    , urlUpdate = urlUpdate
+    , subscriptions = subscriptions
+    }
+
+
+
+-- URL PARSERS - check out evancz/url-parser for fancier URL parsing
+
+
+fromUrl : String -> Maybe Int
+fromUrl url =
+  (String.dropLeft 2 url)
+    |> String.toInt
+    |> Result.toMaybe
+
+
+urlParser : Navigation.Parser (Maybe Int)
+urlParser =
+  Navigation.makeParser (fromUrl << .hash)
+
+
+
+-- MODEL
+
+
+{-| In this example we have two different pages, one for user list and one for
+  selected user's profile.
+-}
+type Page
+  = UserListPage
+  | ProfilePage
+
+
+type alias User =
+  { id : Int
+  , name : String
+  , profilePicture : String
+  }
+
+
+type alias Model =
+  { userList : List User
+  , currentPage : Page
+  , selectedUserId : Int
+  }
+
+
+emptyUser : User
+emptyUser =
+  { id = 0
+  , name = ""
+  , profilePicture = ""
+  }
+
+
+users : List User
+users =
+  [ { id = 0
+    , name = "John"
+    , profilePicture = "http://placekitten.com/400/400"
+    }
+  , { id = 1
+    , name = "Lisa"
+    , profilePicture = "http://placekitten.com/g/400/400"
+    }
+  ]
+
+
+initialModel : Model
+initialModel =
+  { userList = users
+  , currentPage = UserListPage
+  , selectedUserId = 0
```

Comment:
@klemola This is a bit of a lie, isn't it? ;) User with `id=0` (John) is not selected yet.

Shouldn't we use a `Maybe Int` as the type?